core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -55,83 +57,137 @@ object ConsumerGroupCommand {
 
     val consumerGroupService = {
       if (opts.useOldConsumer) {
-        new ZkConsumerGroupService(opts)
+        println("Note: This will only show information about consumers that use ZooKeeper (not those using the Java consumer API).\n")
+        new ZkConsumerGroupService(opts, consumerGroupOutputWriter)
       } else {
+        println("Note: This will only show information about consumers that use the Java consumer API (non-ZooKeeper-based consumers).\n")
         new KafkaConsumerGroupService(opts)
       }
     }
 
     try {
       if (opts.options.has(opts.listOpt))
-        consumerGroupService.list()
-      else if (opts.options.has(opts.describeOpt))
-        consumerGroupService.describe()
+        consumerGroupService.listGroups().foreach { consumerGroupOutputWriter.printMessage(_) }
+      else if (opts.options.has(opts.describeOpt)) {
+        consumerGroupService.describeGroup() match {
+          case None =>
+            consumerGroupOutputWriter.printError(s"The consumer group '${opts.options.valuesOf(opts.groupOpt).asScala.head}' does not exist.")
+          case Some(assignments) =>
+            if (assignments.isEmpty)
+              consumerGroupOutputWriter.printError(s"Consumer group '${opts.options.valuesOf(opts.groupOpt).asScala.head}' is rebalancing.")
+            else
+              consumerGroupOutputWriter.printAssignment(assignments)
+        }
+      }
       else if (opts.options.has(opts.deleteOpt)) {
         consumerGroupService match {
-          case service: ZkConsumerGroupService => service.delete()
-          case _ => throw new IllegalStateException(s"delete is not supported for $consumerGroupService")
+          case service: ZkConsumerGroupService => service.deleteGroups()
+          case _ => throw new IllegalStateException(s"delete is not supported for $consumerGroupService.")
         }
       }
     } catch {
       case e: Throwable =>
-        println("Error while executing consumer group command " + e.getMessage)
-        println(Utils.stackTrace(e))
+        consumerGroupOutputWriter.printError("Executing consumer group command failed due to " + e.getMessage, Some(e))
     } finally {
       consumerGroupService.close()
     }
   }
 
-  sealed trait ConsumerGroupService {
+  case class PartitionAssignmentState(group: String, coordinator: Option[Node], topic: Option[String], partition: Option[Int],
```

Comment:
This `case class` is used in other classes in the same file. Since it's defined outside the scope of those classes `private` wouldn't work, but `protected` would. I hope I didn't misunderstood your point.